woocommerce/woocommerce
# Use Jetpack AI endpoint for product name suggestions

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR replaces the OpenAI API with [Jetpack's completions API](https://github.com/Automattic/jetpack/blob/3be4a6e0afa8b267897dc18afd4e8163754df333/projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-ai.php#L52). Due to the limitations of Jetpack's API (lack of configurable options and chat history), I had to remove the temperature option and the retry functionality from the suggestions service. For some reason, the time to get a response slightly increased.

I added a check to the plugin's bootstrap to confirm that Jetpack is installed and active. If not, we will display an error message on the WP Admin. This has a side effect on the product description feature, too, but since we're [refactoring](https://github.com/woocommerce/woocommerce/pull/38407) it to also use Jetpack's API, it should be fine.

The completions service also checks that Jetpack is connected and returns an error if it is not.

**NOTE**: Currently, Jetpack caches the results of completion for 15 seconds so you might get duplicate results when clicking "Suggest Alternatives", but this will soon change (hence, the `skip_cache` option): https://github.com/Automattic/jetpack/pull/30938

Closes #38451



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

I used my local dev site to test this. But you can use an atomic site too.

1. Install the Jetpack plugin, activate and connect it to WPCOM
2. Add the sticker to your site (pe4Cmx-1cK-p2)
3. Navigate to the product edit page for an existing product that already has a title or description
4. Click on the Get Suggestions button and confirm you get the suggestion for your product name
